Sources/Containerization/ContainerManager.swift

```diff
@@ -0,0 +1,165 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Containerization project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+#if os(macOS)
+
+import ContainerizationError
+import ContainerizationOCI
+import ContainerizationOS
+import Foundation
+
+import Crypto
+
+extension String {
+    fileprivate func hash() throws -> String {
+        guard let data = self.data(using: .utf8) else {
+            fatalError("\(self) could not be converted to Data")
+        }
+        return String(SHA256.hash(data: data).encoded.prefix(36))
+    }
+}
+
+/// A type that manages containers and their resources within a root directory.
+public struct ContainerManager: Sendable {
+    public let imageStore: ImageStore
+    private let vmm: VirtualMachineManager
+
+    var containerRoot: URL {
+        self.imageStore.path.appendingPathComponent("containers")
+    }
+
+    /// Create a new manager with the provided kernel and initfs mount.
+    public init(
+        kernel: Kernel,
+        initfs: Mount,
+    ) throws {
+        self.imageStore = try ImageStore.default
+        try Self.createRootDirectory(path: self.imageStore.path)
+        self.vmm = VZVirtualMachineManager(
+            kernel: kernel,
+            initialFilesystem: initfs,
+            bootlog: self.imageStore.path.appendingPathComponent("bootlog.log").absolutePath()
+        )
+        try Self.createRootDirectory(path: self.imageStore.path)
+    }
+
+    /// Create a new manager with the provided kernel and image reference for the initfs.
+    public init(
+        kernel: Kernel,
+        initfsReference: String
```

Comment:
We might wanna take in an optional boot log path

Reply:
It sets up the bootlog inside the type, into the root directory.

Reply:
Yea that's why I'm honestly okay with this now, as the whole point of it is that everything gets shoved in the dir and can be blown away as one unit